cmd/geth/main.go

```diff
@@ -417,6 +428,35 @@ func attach(ctx *cli.Context) {
 	}
 }
 
+func initGenesis(ctx *cli.Context) {
+	genesisPath := ctx.Args().First()
+	if len(genesisPath) == 0 {
+		utils.Fatalf("must supply path to genesis JSON file")
+	}
+
+	var (
+		datadir = utils.MustMakeDataDir(ctx)
+		cache   = ctx.GlobalInt(utils.CacheFlag.Name)
+		handles = utils.MakeDatabaseHandles()
+	)
+
+	chainDb, err := ethdb.NewLDBDatabase(filepath.Join(datadir, "chaindata"), cache, handles)
+	if err != nil {
+		utils.Fatalf("could not open database: %v", err)
+	}
+
+	genesisFile, err := os.Open(genesisPath)
+	if err != nil {
+		utils.Fatalf("failed to read genesis file: %v", err)
+	}
+
+	block, err := core.WriteGenesisBlock(chainDb, genesisFile)
+	if err != nil {
+		utils.Fatalf("failed to write genesis block: %v", err)
+	}
+	glog.V(logger.Info).Infof("successfully wrote genesis block: %x", block.Hash())
```

Comment:
Since init can also be used to set the rule set, maybe it would make sense to also list that here on the log line? Perhaps also returning it from `core.WriteGenesisBlock`? Although I do know that that would entail in modifying quite some code to return a second param, imho it would be worthwhile in the long run.